nc/include/graphics/GraphicsModule.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include "ecs/component/Camera.h"
+#include "Module.h"
+#include "ui/IUI.h"
+
+namespace nc
+{
+    class Registry;
+
+    /** @brief Interface for general graphics functionality. */
+    struct GraphicsModule : public Module
+    {
+        /**
+         * @brief Set the main Camera.
+         * 
+         * A valid Camera must always be registered during execution
+         * except during scene changes.
+         * 
+         * @param camera A pointer to a valid Camera.
+         */
+        virtual void SetCamera(Camera* camera) noexcept = 0;
+
+        /**
+         * @brief Get the main Camera.
+         * @return Camera*
+         */
+        virtual auto GetCamera() noexcept -> Camera* = 0;
+
+        /**
+         * @brief Set a custom ui to receive draw callbacks during rendering.
+         * @param ui A pointer to a valid IUI implementation.
+         */
+        virtual void SetUi(ui::IUI* ui) noexcept = 0;
```

Comment:
Does graphics work right now if you don't set this? I should test that

Reply:
Yes